pyteal/ast/router.py

```diff
@@ -197,6 +197,16 @@ def __post_init__(self):
                 "https://pyteal.readthedocs.io/en/latest/abi.html#registering-bare-app-calls"
             )
 
+    def asdict(self) -> dict[str, OnCompleteAction]:
```

Comment:
nit: this is a getter method, so we can consider making it `@property`?

Reply:
The pattern of `asdict()` being a method is widespread in the Python world, so I'm inclined to stay with this approach. However, your suggestion is a reasonable one and we do use `@property`'s in several other places, for more custom properties such as `TxnObject.assets`. It might be worthwhile synthesizing when we should/shouldn't use properties and add it to our [code guidelines](https://github.com/algorand/pyteal/blob/005d8a618b652adca90f413dc0ffae303615e981/CONTRIBUTING.md#L19).

Reply:
ok, I am just suggesting a potential move, but I don't want to catch too much attention on code guidelines, so I am resolving here.